internal/security/proxy/tomlParser.go

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package proxy
+
+import (
+	"fmt"
+	"github.com/BurntSushi/toml"
+)
+
+type tomlConfig struct {
+	Title         string
+	KongURL       kongurl
+	KongAuth      kongauth
+	KongACL       KongACLPlugin
+	SecretService secretservice
+	EdgexServices map[string]service
+}
+
+type kongurl struct {
+	Server             string
+	AdminPort          string
+	AdminPortSSL       string
+	ApplicationPort    string
+	ApplicationPortSSL string
+}
+
+type kongauth struct {
+	Name       string
+	TokenTTL   int
+	Resource   string
+	OutputPath string
+}
+
+type kongacl struct {
+	Name      string
+	WhiteList string
+}
+
+type secretservice struct {
+	Server          string
+	Port            string
+	HealthcheckPath string
+	CertPath        string
+	TokenPath       string
+	CACertPath      string
+	SNIS            string
+}
+
+type service struct {
+	Name     string
+	Host     string
+	Port     string
+	Protocol string
+}
+
+func LoadTomlConfig(path string) (*tomlConfig, error) {
+	config := tomlConfig{}
+	_, err := toml.DecodeFile(path, &config)
+	return &config, err
+}
+
+func (cfg *tomlConfig) GetCertPath() string {
+	return cfg.SecretService.CertPath
+}
+
+func (cfg *tomlConfig) GetTokenPath() string {
```

Comment:
> also the tests failed because `internal/security/proxy/consumer.go` is not formatted with gofmt:
> 
> ```
> 16:18:56 gofmt -l .
> 16:18:57 internal/security/proxy/consumer.go
> 16:18:57 [ "`gofmt -l .`" = "" ]
> 16:18:57 make: *** [test] Error 1
> 16:18:57 Build step 'Execute shell' marked build as failure
> ```

somehow gofmt -l . didn't format locally so I had to run gofmt -s -w .  instead. Fixed.